src/handlers/guild.ts

```diff
@@ -622,3 +629,148 @@ export function getGuild(guildID: string, counts = true) {
     { with_counts: counts },
   ) as Promise<UpdateGuildPayload>;
 }
+
+/** Returns the guild template if it exists */
+export function getGuildTemplate(
+  guildID: string,
+  templateCode: string,
+): Promise<Template> {
+  return RequestManager.get(
+    `${endpoints.GUILD_TEMPLATES(guildID)}/${templateCode}`,
+  ) as any;
+}
+
+/**
+ * Create a new guild based on a template
+ * NOTE: This endpoint can be used only by bots in less than 10 guilds.
+ */
+export async function createGuildFromTemplate(
+  templateCode: string,
+  data: CreateGuildFromTemplate,
+) {
+  if (cache.guilds.size >= 10) {
+    throw new Error(
+      "This function can only be used by bots in less than 10 guilds.",
+    );
+  }
+
+  if (data.icon) {
+    data.icon = await urlToBase64(data.icon);
+  }
+
+  try {
```

Comment:
I don't understand this try catch. There is no point of a try catch that just throws the error again. That just makes this code slower.

```ts
return RequestManager.post() as Promise<CreateGuildPayload>
```